src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java

```diff
@@ -43,26 +42,8 @@ class FileImportControl extends AbstractImportControl {
      */
     /* package */ FileImportControl(PkgImportControl parent, String name, boolean regex) {
         super(parent, MismatchStrategy.DELEGATE_TO_PARENT);
```

Comment:
This is the constructor. so after loading all the configuration and setup when `visitToken` in ImportControl
https://github.com/checkstyle/checkstyle/blob/83788d6ff5fb659f8590906557c46b79a018f21e/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java#L237
after executing this statement 
https://github.com/checkstyle/checkstyle/blob/83788d6ff5fb659f8590906557c46b79a018f21e/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java#L80
after this method 
https://github.com/checkstyle/checkstyle/blob/83788d6ff5fb659f8590906557c46b79a018f21e/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java#L90-L102

where we already check that if regex is true or not so that's why replacing this https://github.com/checkstyle/checkstyle/blob/83788d6ff5fb659f8590906557c46b79a018f21e/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java#L48-L55
if as true will not make any affect. https://github.com/checkstyle/checkstyle/blob/83788d6ff5fb659f8590906557c46b79a018f21e/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java#L33
The only usage of this field is at https://github.com/checkstyle/checkstyle/blob/83788d6ff5fb659f8590906557c46b79a018f21e/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java#L95-L97
so we can remove this field  and directly initialize that in method it will not make any issue